### ISSUE TITLE
Support for printing the version of the application

### ISSUE THREAD

**Issue by jorisroovers** [jawher/mow.cli#18]
This is a feature request for providing some syntactic sugar to easily print the version of a cli application. A ```--version``` option is a widely used by commandline programes.

Currently, I use the following workaround:
```go
cp := cli.App("mycli", "My cli description")
cp.Command("version", "Prints the version", func(cmd *cli.Cmd) {
	cmd.Action = func() {
		fmt.Println("mycli, version", "0.1.0")
	}
})
```
It would however be nice to just be able to do:
```go
// notice the extra "0.1.0" parameter for cli.App
cp := cli.App("mycli", "My cli description", "0.1.0")
```
This version string would then be printed automatically when users invoke: ```mycli version``` or the option variants such as as ```mycli --version```, ```mycli -v```.

Ideally these commands are automatically disabled for cli apps that don't specify a version.

You could conceivable also print the version number as part of the help output, but as far as I'm concerned that'd be a choice for the implementer.

**Comment by jawher** [comment]
@jorisroovers Thanks for bringing this issue up.

I agree that supporting the `--version` idiom is needed.
Handling it correctly requires you to:

* Declare the boolean flag version on the app
* Adding an action to the app even when the app only exposes commands
* In the said action, print the version if the flag is set, or print the usage help text otherwise

Which is cumbersome.

Handling the version via a command (like in the go or docker commands) is much simpler, as you've shown in your example.
Besides, such a command usually shows more info (daemon and client version, etc.), and so can require executing some code instead of just printing a constant.
That's why I don't think adding a special support for a `version` command is really needed at this time.

Back to the version flag, I'm thinking about something like:

```go
app.Version("v version", "mycli 1.0.7-beta3, git: 5a708d")
```

where the first param is as in the other options, i.e. option names (short and long)), and the second param is the value to print.

@jorisroovers What do you think ?

> You could conceivable also print the version number as part of the help output, but as far as I'm concerned that'd be a choice for the implementer.

This can be achieved by simply including the version in the app description, e.g.:

```go
cp := cli.App("mycli", "My cli description (version: 1.0.4)")
```

**Comment by jorisroovers** [comment]
I agree that ```mycli --version``` is the widespread idiom and preferable over ```my cli version``` (although funny enough, the go cli itself uses ```version``` instead of ```--version```). If it were cleaner to implement in the current version of mow.cli, then that's what I would've done as well.

The suggested approach ```app.Version("v version", "mycli 1.0.7-beta3, git: 5a708d")```  sounds perfect to me.

Are you tackling this? I can also take a stab at it if you want.

**Comment by jawher** [comment]
@jorisroovers : By all means, if you're willing to tackle this, I'd highly appreciate it :)

**Comment by jorisroovers** [comment]
@jawher Ok, on it :-)

**Comment by jawher** [comment]
Implemented in #20.
